File: Niveau 3/Niveau 3.7/péremption.py
#-------------------------------------------------------------------------------
# Name:        Dates de péremption
# Purpose:     France-IOI niveau 3.7
#
# Author:      Sébastien
#
# Created:     25/09/2022
#-------------------------------------------------------------------------------

def main():
    pile = []
    o = int(input())

    for _ in range(o):
        n, date = map(int, input().split())
        if n > 0:
            for _ in range(n):
                pile.append(date)
        else:
            pile = pile[:n]

    print(min(pile))


if __name__ == '__main__':
    main()

"""
8
3 20040810
-1 0
-1 0
3 20040920
-1 0
4 20040916
-3 0
-2 0
"""

File: Niveau 3/Niveau 3.7/péremption.hs
--------------------------------------------------------------------------------
-- Name:        Dates de péremption
-- Purpose:     France-IOI niveau 3.7
--
-- Author:      Sébastien
--
-- Created:     25/09/2022
--------------------------------------------------------------------------------

import Control.Monad
import Data.List

traiter :: Int -> [Int] -> IO [Int]
traiter 0 pile = return pile
traiter o pile = do
    pile' <- traiter (o-1) pile
    (n:date:_) <- fmap (map read . words) getLine :: IO [Int]
    -- putStrLn $ "Opération #" ++ show o ++ ": nombre = " ++ show n ++ ", date = " ++ show date
    return $ if n > 0 then (replicate n date) ++ pile' else drop (-n) pile'

main :: IO()
main = do  
    -- Réalise les opérations sur la pile
    o <- readLn :: IO Int
    pile <- traiter o []
    
    -- Affiche le stock final
    print $ minimum pile
